@here/harp-mapview/lib/SphereHorizon.ts

```diff
@@ -0,0 +1,333 @@
+//  .~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"~}:::::::::::::::::~~~~~~~""""":},,,,,,:!!:
+//  ,~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~!}:::::::::::::::::::~~~~~~~~~~~},,,,,",,,:!!~
+// `~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~|{::::::::::::::::::::::~~~~~~~~},,,",,,,,,",!!
+// .~""""""~~~~~~~~~~~~~~~~~~~~"~~~~~~~(/::::::::::::::::::::::::::::::},",,,,,,,,,,,"!!   ^
+// .~""""""""""""~~~~~~~~~~~~~~~~"~~~~~{|::::::::::::::)^::::::::::::::},",,,,,,,,"""," !! |
+// .~~~~~~~~~~~""""""""~~~~~~~~~~~~""~~{|:::::!!!!!!!!!}\!!!!!!!!!!!!!!$!!!!!!!!!!!!!!!!!!!E---->
+// .~""""""~"""",,"""""""""~~~~~~~~~~""(/::::::::::::::::::::::::::::::}"~""""~~~~~~~~~'
+// `~~"~~""",,,,,,,,,,,,,,""""~~~~~~~~~\{::::::::::::::::::::::::::::::}~~~~~~"""""""~"'
+//  ,,,,,,,,,,,,,,,,,,,,,,,,,"""""~~~~~!}::::::::::::::::::::::::::::::}~~~~~~~~~~~~~~~`
+//  ."",,,,,,,,,,,,,,,,,,,,,,,,,,""~~~~~}::::::::::::::::::::::::::::::}~~~~~~~~~~~~~~,
+//   ,",,,,,,,,,,,,,,,,,,,,,,,,,,,,""""~}:::::::::::::::::::::::::::::|{~~~~~~~~~~~~~~`
+//   `",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,""~}:::::::::::::::::::::::::::::}!~~~~~~~~~~~~~'
+//    '"",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,">)::::::::::::::::::::::::::::}~~~~~~~~~~~~~"
+//     '",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,""}:::::::::::::::::::::::::::!}~~~~~~~~~~~~~`
+//      '",,,,,,,,,,,,,,,,,,,,,,,,,,,",,,}!::::::::::::::::::::::::::}!~~~~~~~~~~~"`
+//       '"",,,,,,",,,,,,,,,,,,,,,,,,,,,"!}::::::::::::::::::::::::::}:~~~~~~~:~~,`
+//        `,"",,,,"",,,,,,,,,,,,,,,,,,,,""}^::::::::::::::::::::::::}+~~~~~~~:::,
+//         `'",,,,,,""",,,,,,,,,,,,,,,,,"":$:::::::::::::::::::::::^}~:~~~~~~~~.
+//           `,",,,,""""",,,,,,,,,,,,,,,,,"?(::::::::::::::::::::::}:~~~~~~~~'
+//             .,"",,,"","",,,,,,,,,,,,,,,""}*::::::::::::::::::::};~~~~:~~'`
+//               `,""",,,,,,,,,,,,,,,,,,,,,,"}>::::::::::::::::::{+:~~:~"'
+//                 `.,"",,,,,,,,,,,,,,,,,,,,"~{\::::::::::::::::}*~~~~'`
+//                    `.,""",,,,,,,,,,,,,,,""~~|{:::::::::::::^}!~"'.
+//                       `.',"",,,,,,,,,,,,"""~~:{(:::::::::^{\'.`
+//                            `.',,"""""",,""""~~~:////?|\|>:`
+//                                  ```...''''''''..```.`
+//  *
+ *
+ * @internal
+ */
+export class SphereHorizon {
+    private readonly m_matrix: THREE.Matrix4;
+    private readonly m_radius: number;
+    private readonly m_normalToTangentAngle: number;
+    private readonly m_distanceToHorizonCenter: number;
+    private readonly m_intersections: number[][] = [];
+    private m_cameraPitch?: number;
+    private m_hFovVertical?: number;
+    private m_hFovHorizontal?: number;
+
+    /**
+     * Constructs the SphereHorizon for the given camera and projection.
+     *
+     * @param m_camera - The camera used as a reference to compute the horizon.
+     * @param m_projection - A spherical projection.
+     */
+    constructor(
+        private readonly m_camera: THREE.PerspectiveCamera,
+        private readonly m_projection: Projection
```

Comment:
Will this work for any other projection, then the sphereProjection? If not, there is no need, to pass the projection, but just use sphereProjection

Reply:
In theory it works for any Sphere projection (that is any instance of Sphere projection with any radius, not necessarily the earth radius). In practice, it's true that we only use the sphere projection "singleton" with the earth radius.

Reply:
actually i lied, since I'm using the earth radius constant everywhere.

Reply:
I've removed the projection.